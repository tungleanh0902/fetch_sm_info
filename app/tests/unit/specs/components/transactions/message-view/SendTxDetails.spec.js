import { shallowMount } from "@vue/test-utils"
import SendTxDetails from "src/components/transactions/message-view/SendTxDetails"

describe(`SendTxDetails`, () => {
  let wrapper

  const tx = {
    type: "SendTx",
    hash: "3CA728671B8078E71697B62237AD694052779F80B56880F6A6F1702F53EA3081",
    height: 308453,
    timestamp: "2020-01-10T09:02:54Z",
    memo: "",
    success: true,
    fees: [
      {
        denom: "ATOM",
        amount: "0.00045",
      },
    ],
    details: {
      from: ["cosmosvaloper123"],
      to: ["cosmosvaloper456"],
      amount: {
        denom: "ATOM",
        amount: "10",
      },
    },
  }

  it(`renders a sent transaction message`, () => {
    wrapper = shallowMount(SendTxDetails, {
      propsData: {
        transaction: tx,
        sessionAddress: "cosmosvaloper123",
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`renders a received transaction message`, () => {
    wrapper = shallowMount(SendTxDetails, {
      propsData: {
        transaction: tx,
        sessionAddress: "cosmosvaloper456",
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
