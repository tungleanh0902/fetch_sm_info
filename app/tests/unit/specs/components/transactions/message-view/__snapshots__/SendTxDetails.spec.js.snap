// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendTxDetails renders a received transaction message 1`] = `
<div
  class="tx__content"
>
  <transactionicon-stub
    transactiontype="Received"
  />
   
  <div
    class="tx__content__left"
  >
    <h3>
      Received
    </h3>
     
    <span>
      From 
    </span>
     
    <address-stub
      address="cosmosvaloper123"
      tooltiptext=""
    />
  </div>
   
  <div
    class="tx__content__right"
  >
    <p
      class="amount"
    >
      
      10 
      ATOM
    
    </p>
  </div>
</div>
`;

exports[`SendTxDetails renders a sent transaction message 1`] = `
<div
  class="tx__content"
>
  <transactionicon-stub
    transactiontype="Sent"
  />
   
  <div
    class="tx__content__left"
  >
    <h3>
      Sent
    </h3>
     
    <span>
      To 
    </span>
     
    <address-stub
      address="cosmosvaloper456"
      tooltiptext=""
    />
  </div>
   
  <div
    class="tx__content__right"
  >
    <p
      class="amount"
    >
      
      10 
      ATOM
    
    </p>
  </div>
</div>
`;
